File: Design_Pattern_and_Principles/StrategyPatternExample/PaymentContext.java
public class PaymentContext {
    private PaymentStrategy strategy;

    public void setPaymentStrategy(PaymentStrategy strategy) {
        this.strategy = strategy;
    }
    public void checkout(double amount) {
        if (strategy == null) {
            System.out.println("Error: No strategy selected.");
            return;
        }
        strategy.pay(amount);
    }
}

File: Week_1/Design_Pattern_and_Principles/StrategyPatternExample/PaymentContext.cs
using System;

public class PaymentContext {
    private IPaymentStrategy strategy;

    public void SetPaymentStrategy(IPaymentStrategy strategy) {
        this.strategy = strategy;
    }

    public void Checkout(double amount) {
        if (strategy == null) {
            Console.WriteLine("Error: No strategy selected.");
            return;
        }
        strategy.Pay(amount);
    }
}